### PR TITLE
Added network stats

### PR DESCRIPTION
**Pull request by essigt**
Added stats from /proc/net/dev

image

### CODE REVIEW

**Comment by BenoitHanotte** [comment]
Hello @essigt, thanks for your PR, this has been on my todo list for some time but I never got around to work on it :)
My only concern about it is that, as far as I know, there is no way to get network usage for the process tree only and we can only get it for the entire host? (it is not an issue, I would just make sure that it is clearly stated and differentiated from the IO graphs that show the usage only on the process tree). Also, is there a risk of capturing local/virtual network interfaces that could make the result invalid?
I will run some tests today or early next week to make sure it works on our infra and then we should be able to merge :)

**Comment by essigt** [comment]
Hello @BenoitHanotte I committed a change to only take non-virtual interfaces into account and also stated the fact more clearly that the graph shows the IO from the entire host.

File: babar-agent/src/main/java/com/criteo/babar/agent/profiler/utils/ProcFSUtils.java [hunk 1]
```diff
@@ -174,7 +174,55 @@ protected static ProcPidIO parsePidIO(String output) {
...
+                rxBytes += Long.parseLong(cols[1]);
+                txBytes += Long.parseLong(cols[9]);
+            }
+        }
+
+        return new ProcNetIO(rxBytes, txBytes);
+	}
+
+	protected static List<String> parseRealDevices(String outputSysClassNet) {
+		List<String> realDevices = new LinkedList<>();
```
Comment (BenoitHanotte) [inline babar-agent/src/main/java/com/criteo/babar/agent/profiler/utils/ProcFSUtils.java]: I see that we then check for a device using `contains()`, for this reason I would avoid using a `LinkedList` and instead use an `ArrayList` (which will be better for cache locality and will avoid having to allocate an iterator to go through the list).
Even better, as we will usually have more than a 2-3 devices, we could use a `HashSet` to allow searching in `O(1)`

File: babar-agent/src/main/java/com/criteo/babar/agent/profiler/utils/ProcFSUtils.java [hunk 2]
```diff
@@ -174,7 +174,55 @@ protected static ProcPidIO parsePidIO(String output) {
...
+		String[] lines = outputSysClassNet.split("\n");
+		
+		for(String line : lines) {
+			String[] cols = line.split(" +");
+			
+			if(cols.length == 11) {
+				if(!cols[10].startsWith("../../devices/virtual/net/")) {
```
Comment (BenoitHanotte) [inline babar-agent/src/main/java/com/criteo/babar/agent/profiler/utils/ProcFSUtils.java]: the two `if`s could be factorized as
`if (cols.length == 11 && !cols[10].startsWith("../../devices/virtual/net/")`

File: babar-report/src/IO.vue [hunk 3]
```diff
@@ -9,6 +9,10 @@
     <template v-if="isProcFSProfilerUsed">
     
       <b-container fluid>
+        <PlotTimeSeries title="Network bytes received/transmitted / sec" yAxis="bytes/sec" :series="series.netIO" />
+        <div class="explanation">
+          This graph shows the total number of bytes of data transmitted or received by all non-virtual interfaces by the entire process-tree as reported by the <kbd>/prod/net/dev</kbd> file.<br>
```
Comment (BenoitHanotte) [inline babar-report/src/IO.vue]: There's a typo, it's `/proc`. I have made that typo everywhere else, I'll push a PR to fix the other ones ;)

**Review by BenoitHanotte — CHANGES_REQUESTED** [review]
Looks good, there are only a few minor comments :+1:

**Comment by BenoitHanotte** [comment]
I was also wondering whether we wanted to show both the median and max network usage instead of only the maximum one that I believe is currently shown?
But that can be adjusted later so no worries

**Comment by essigt** [comment]
Hey @BenoitHanotte thanks for the great feedback :)
I try to add the changes in the next days.

File: babar-report/src/IO.vue [hunk 4]
```diff
@@ -11,7 +11,7 @@
       <b-container fluid>
         <PlotTimeSeries title="Network bytes received/transmitted / sec" yAxis="bytes/sec" :series="series.netIO" />
         <div class="explanation">
-          This graph shows the total number of bytes of data transmitted or received by all non-virtual interfaces by the entire process-tree as reported by the <kbd>/prod/net/dev</kbd> file.<br>
+          This graph shows the total number of bytes of data transmitted or received by all non-virtual interfaces by the entire process-tree as reported by the <kbd>/proc/net/dev</kbd> file.<br>
```
Comment (BenoitHanotte) [inline babar-report/src/IO.vue]: I may have missed this one on the first pass, but if I am not mistaken it is not the network usage of the process tree but the one of the entire system (the host OS)?

Comment (essigt) [reply babar-report/src/IO.vue]: Yes of course, it is the network usage of the entire system

**Review by BenoitHanotte — APPROVED** [review]

**Comment by BenoitHanotte** [comment]
@essigt Thanks a lot for the PR, that's a great addition, I am merging it right now :)

**Pull request merged**